network-api/networkapi/wagtailpages/tests/buyersguide/test_article_page.py

```diff
@@ -206,3 +210,114 @@ def test_secondary_related_articles_no_related_articles(self):
+                    "slug": article_page.slug,
+                    "search_description": article_page.search_description,
+                    "search_image": None,
+                }
+            ),
+        )
+
+        self.assertEqual(1, len(test_form.errors))
+        self.assertIn("search_image", test_form.errors)
+        self.assertIn("This field is required.", test_form.errors["search_image"])
+
+    def test_article_page_requires_search_description(self):
+        """
+        Test that the buyersguide article page form will
+        return a validation error if no "search_description" is set.
+        """
+        article_page = buyersguide_factories.BuyersGuideArticlePageFactory(parent=self.content_index)
+
+        test_form = self.article_page_form(
+            instance=article_page,
+            data=self.generate_form_data(
+                {
+                    "title": article_page.title,
+                    "slug": article_page.slug,
+                    "search_description": None,
+                    "search_image": article_page.search_image,
+                }
+            ),
+        )
+
+        self.assertEqual(1, len(test_form.errors))
+        self.assertIn("search_description", test_form.errors)
+        self.assertIn("This field is required.", test_form.errors["search_description"])
+
+    def test_article_page_requires_both_search_fields(self):
+        """
+        Test that the buyersguide article page form will return validation
+        errors for both "search_image" and "search_description" fields
+        if neither are updated.
+        """
+        article_page = buyersguide_factories.BuyersGuideArticlePageFactory(parent=self.content_index)
+
+        test_form = self.article_page_form(
+            instance=article_page,
+            data=self.generate_form_data(
+                {
+                    "title": article_page.title,
+                    "slug": article_page.slug,
+                    "search_description": None,
+                    "search_image": None,
+                }
+            ),
+        )
+
+        self.assertEqual(2, len(test_form.errors))
+        self.assertIn("search_image", test_form.errors)
+        self.assertIn("search_description", test_form.errors)
+        self.assertIn("This field is required.", test_form.errors["search_image"])
+        self.assertIn("This field is required.", test_form.errors["search_description"])
+
+    def test_article_page_with_search_fields_is_valid(self):
+        """
+        Test that a buyersguide article page form with
+        the search fields set is valid with no errors.
+        """
+        article_page = buyersguide_factories.BuyersGuideArticlePageFactory(parent=self.content_index)
+
+        test_form = self.article_page_form(
+            instance=article_page,
+            data=self.generate_form_data(
+                {
+                    "title": "New Test Title",
+                    "slug": "new-test-title",
```

Comment:
This should not be the error, but why are you changing slug and title here?

Reply:
Hi Tibor, another good catch, originally I added the title/slug fields in the test form data as they are also required fields, and leaving them out would return a validation error.

In this test, I was updating the title and slug as I believed this would be a good test of "Made a change in the form, but have valid data for the `search_image` and `search_description`", but I think that this makes the test a little hard to follow.

Instead, I have updated `generate_form_data()` to add the page's `title`,`slug`, and required `body` field to the form data, as well a comment explaining that these are required and necessary to add to the form, so we dont get any validation errors against fields we are not trying to test against.